test/new-e2e/tests/discovery/linux_test.go

```diff
@@ -161,6 +320,100 @@ func assertRunningCheck(t *assert.CollectT, remoteHost *components.RemoteHost, c
 	assertCollectorStatusFromJSON(t, statusOutput, check)
 }
 
+// assertNotRunningCheck asserts that the given process agent check is not running
+func assertNotRunningCheck(t *assert.CollectT, remoteHost *components.RemoteHost, check string) {
+	statusOutput := remoteHost.MustExecute("sudo datadog-agent status collector --json")
+	var status collectorStatus
+	err := json.Unmarshal([]byte(statusOutput), &status)
+	require.NoError(t, err, "failed to unmarshal agent status")
+	assert.NotContains(t, status.RunnerStats.Checks, check)
+}
+
+func assertProcessServiceDiscoveryData(t *testing.T, c *assert.CollectT, procs []*agentmodel.Process, expectedLanguage agentmodel.Language, expectedPortInfo *agentmodel.PortInfo, expectedServiceDiscovery *agentmodel.ServiceDiscovery) {
```

Comment:
Thought: This function is quite long and was hard to read for me. You can consider breaking it up into helpers for each check that you do.